modules/transactionpool/transactionpool.go

```diff
@@ -107,24 +74,33 @@ func New(cs modules.ConsensusSet, g modules.Gateway) (tp *TransactionPool, err e
 		consensusSet: cs,
 		gateway:      g,
 
-		transactions:   make(map[crypto.Hash]struct{}),
-		siacoinOutputs: make(map[types.SiacoinOutputID]types.SiacoinOutput),
-		fileContracts:  make(map[types.FileContractID]types.FileContract),
-		siafundOutputs: make(map[types.SiafundOutputID]types.SiafundOutput),
+		knownObjects:        make(map[ObjectID]struct{}),
+		transactionSets:     make(map[TransactionSetID][]types.Transaction),
+		transactionSetDiffs: make(map[TransactionSetID]modules.ConsensusChange),
 
-		referenceSiacoinOutputs:        make(map[types.SiacoinOutputID]types.SiacoinOutput),
-		referenceFileContracts:         make(map[types.FileContractID]types.FileContract),
-		referenceFileContractRevisions: make(map[crypto.Hash]types.FileContract),
-		referenceSiafundOutputs:        make(map[types.SiafundOutputID]types.SiafundOutput),
+		// The consensus change index is intialized to '-1', which indicates
+		// that no consensus changes have been sent yet. The first consensus
+		// change will then have an index of '0'.
+		consensusChangeIndex: -1,
 
-		mu: sync.New(modules.SafeMutexDelay, 1),
+		mu: sync.New(modules.SafeMutexDelay, 5),
 	}
 
 	// Register RPCs
-	g.RegisterRPC("RelayTransaction", tp.RelayTransaction)
+	g.RegisterRPC("RelayTransactionSet", tp.RelayTransactionSet)
```

Comment:
unfortunately the internals of the transaction pool changed dramatically enough that retaining support for the old call would be harmful.

The problem is that on the old system, you would submit a parent and then submit the child. On the new system, once you submit a parent you /cannot/ submit a child. It'll be rejected. This could cause weird problems and lost coins.

There might be a way to have the new tpool automatically detect if an incoming transaction is a child to an existing known transaction set, and then extend the transaction set. But there are lots of corner cases involved with doing that, and I think the code complexity doesn't really justify the use-case.